src/server/src/main/java/io/cassandrareaper/jmx/JmxConnectionFactory.java

```diff
@@ -118,7 +118,7 @@ protected JmxProxy connect(
     }
   }
 
-  public final JmxProxy connect(Node node, int connectionTimeout)
+  public JmxProxy connect(Node node, int connectionTimeout)
```

Comment:
why unfinalise the method?

Reply:
That's actually how many of the tests are built and overriding those methods is necessary.
Due to how the JmxConnectionFactory is injected in the context, I couldn't use a mock instead (as far as I can remember).